core/sr-primitives/src/generic/unchecked_extrinsic.rs

```diff
@@ -125,15 +125,14 @@ where
 	fn encode(&self) -> Vec<u8> {
 		let mut v = Vec::new();
 
-		// need to prefix with the total length as u32 to ensure it's binary comptible with
-		// Vec<u8>. we'll make room for it here, then overwrite once we know the length.
-		v.extend(&[0u8; 4]);
-
 		self.signature.encode_to(&mut v);
 		self.function.encode_to(&mut v);
 
-		let length = (v.len() - 4) as u32;
-		length.using_encoded(|s| v[0..4].copy_from_slice(s));
+		// need to prefix with the total length to ensure it's binary comptible with
+		// Vec<u8>.
+		let mut length: Vec<()> = Vec::new();
+		length.resize(v.len(), ());
+		length.using_encoded(|s| { v.splice(0..0, s.iter().cloned()); });
```

Comment:
will always need to do a `memmove`. we could skip this in most cases by a well chosen space reservation as before.